app/concepts/tickets/use_cases/create_for_reservation.rb

```diff
@@ -0,0 +1,43 @@
+module Tickets
+  module UseCases
+    class CreateForReservation
+      SeatsNotAvailableError = Class.new(StandardError)
+
+      attr_reader :reservation, :screening, :tickets_params
+
+      def initialize(reservation:, screening:, tickets_params: [])
+        @tickets_params = tickets_params
+        @reservation = reservation
+        @screening = screening
+      end
+
+      def call
+        Ticket.transaction do
```

Comment:
@wmaciejak is it a correct way to raise any bugs?

Reply:
Yeah, it's an awesome example of how to use it! ❤️ 

For the next step of refactorization, consider that lines 16 and 17 looks like something that might be named generally as `validation` cause you can check seats availability before ticket's creations. Try to extract this validation to a separated class, for instance `validator` in ticket concept. 

Anyway, this is an idea for the future, please, first of all, finish the current scope of monte way and then refactor.